### PR TITLE
chore(custom-resources): running Python tests from Lerna

### PR DESCRIPTION
**Pull request by sakurai-ryo**
Currently, the Python test code used in `custom-resource-handlers` is not included in the execution target of the `lerna run test` command and is not executed by CI.
This PR fixes the build process and the broken tests.

To run Python tests from Lerna, we need to run the shell script to run the Python tests in the `test` section of npm scripts.
It was originally executed using the post script of npm scripts.
https://github.com/aws/aws-cdk/pull/971/files#diff-a683107cab7bc3ac616d2db6e4e02c4f2aa677c56e45f0d8f822d82929a83d7dR41

The `test` section of `npm scripts` in the `@aws-cdk/custom-resource-handlers` package currently runs the `jest` command, which is used to run all tests in the package and specific tests.
I created a new shell script and branched out for these two use cases.

I am unfamiliar with Lerna or the CDK build process, so I would appreciate feedback from anyone who sees this PR.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/custom-resource-handlers/test/aws-s3-deployment/bucket-deployment-handler/test.py [hunk 1]
```diff
@@ -576,7 +576,7 @@ def test_fails_when_physical_id_not_present_in_update(self):
             "DestinationBucketName": "<dest-bucket-name>",
         }, expected_status="FAILED")
 
-        self.assertEqual(update_resp['Reason'], "invalid request: request type is 'Update' but 'PhysicalResourceId' is not defined")
+        self.assertEqual(update_resp['Reason'], "invalid request: request type is '{'Update'}' but 'PhysicalResourceId' is not defined")
```
Comment (sakurai-ryo) [inline packages/@aws-cdk/custom-resource-handlers/test/aws-s3-deployment/bucket-deployment-handler/test.py:579]: Changed in this PR, but the test code has not been modified.
https://github.com/aws/aws-cdk/pull/28057

File: packages/@aws-cdk/custom-resource-handlers/test/aws-s3/notifications-resource-handler/test_index.py [hunk 2]
```diff
@@ -62,10 +62,7 @@ def make_eventbridge_configuration():
     return { EVENTBRIDGE_CONFIGURATION: {} }
```
Comment (sakurai-ryo) [inline packages/@aws-cdk/custom-resource-handlers/test/aws-s3/notifications-resource-handler/test_index.py:62]: This PR changed this file, but the test code did not.
https://github.com/aws/aws-cdk/pull/28132

File: packages/@aws-cdk/custom-resource-handlers/test/aws-s3/notifications-resource-handler/test_index.py [hunk 3]
```diff
@@ -86,337 +83,330 @@ def merge_notification_configurations(conf1: Dict, conf2: Dict):
 
 
 class ManagedBucketTest(unittest.TestCase):
-    @patch("index.put_bucket_notification_configuration")
+    @patch('index.s3')
```
Comment (sakurai-ryo) [inline packages/@aws-cdk/custom-resource-handlers/test/aws-s3/notifications-resource-handler/test_index.py:86]: There was a function that executes the `put_bucket_notification_configuration` method of `boto3`, but it has been removed.

**Review by TheRealAmazonKendra — APPROVED** [review]
Oh, wow. This is a very very long term oversight on our part. We shouldn't have unit tests that aren't jest anymore, but taking this change is obviously much better than having tests that don't run. Thanks for catching this.

**Comment by TheRealAmazonKendra** [comment]
@Mergifyio update

**Review by TheRealAmazonKendra — APPROVED** [review]

**Comment by TheRealAmazonKendra** [comment]
Looks like some of the test updates caused a failure. Can you please take a look?

**Pull request closed without merging**